Data/Hashable.hs

```diff
@@ -251,3 +258,23 @@ instance (IsString a, Hashable a) => IsString (Hashed a) where
 instance Foldable Hashed where
   foldr f acc (Hashed a _) = f a acc
 
+-- instances for @Data.Functor.Classes@ higher rank typeclasses
+-- in base-4.9 and onward.
+#if MIN_VERSION_base(4,9,0)
+instance Eq1 Hashed where
+  liftEq f (Hashed a ha) (Hashed b hb) = ha == hb && f a b
+
+instance Ord1 Hashed where
+  liftCompare f (Hashed a _) (Hashed b _) = f a b
+
+instance Show1 Hashed where
+  liftShowsPrec sp _ d (Hashed a _) = showsUnaryWith sp "hashed" d a
+#endif
+
+-- This function is copied from Data.Functor.Classes, which does
+-- not export it.
+showsUnaryWith :: (Int -> a -> ShowS) -> String -> Int -> a -> ShowS
```

Comment:
> This function is copied from Data.Functor.Classes, which does not export it.

[Doesn't it?](http://hackage.haskell.org/package/base-4.9.0.0/docs/Data-Functor-Classes.html#v:showsUnaryWith)